File: Dddml.Wms.HttpServices.PhpClient/Wms/Domain/AttributeValueIdFlattenedDto.php
<?php

namespace Wms\Domain;

use JMS\Serializer\Annotation\Type;

class AttributeValueIdFlattenedDto
{

    const PROPERTIES = [
            'attributeId' => 'string',
            'value' => 'string',
        ];

    public static function getPropertyNames()
    {
        return array_keys(static::PROPERTIES);
    }

    public static function getPropertyTypes()
    {
        return array_values(static::PROPERTIES);
    }

	
    /**
     * @var AttributeValueId
     */
    private $value;

    /**
     * @param AttributeValueId $value
     */
    public function __construct(AttributeValueId $value = null)
    {
        if ($value) {
            $this->value = $value;
        } else {
            $this->value = new AttributeValueId();
        }
    }

    /**
     * @return AttributeValueId
     */
    public function toAttributeValueId()
    {
        return $this->value;
    }

    /**
     * @return string
     */
    public function getAttributeId()
    {
        return $this->value->getAttributeId();
    }

    /**
     * @param string $attributeId
     */
    public function setAttributeId($attributeId)
    {
        $this->value->setAttributeId($attributeId);
    }

    /**
     * @return string
     */
    public function getValue()
    {
        return $this->value->getValue();
    }

    /**
     * @param string $value
     */
    public function setValue($value)
    {
        $this->value->setValue($value);
    }


}

File: Dddml.Wms.HttpServices.JavaClient/src/org/dddml/wms/domain/AttributeValueIdFlattenedDto.java
package org.dddml.wms.domain;

import java.util.HashMap;
import java.util.Map;

public class AttributeValueIdFlattenedDto
{


    public static final String[] propertyNames = new String[]{
            "attributeId",
            "value",
        };

    public static final String[] propertyTypes = new String[]{
            "String",
            "String",
        };

    public static final Map<String, String> propertyTypeMap;

    static {
        propertyTypeMap = new HashMap<String, String>();
        for (int i = 0; i < propertyNames.length; i++ ) {
            propertyTypeMap.put(propertyNames[i], propertyTypes[i]);
        }
    }
	
    private AttributeValueId value;

    public AttributeValueIdFlattenedDto()
    {
        this(new AttributeValueId());
    }

    public AttributeValueIdFlattenedDto(AttributeValueId value)
    {
        this.value = value;
    }

    public AttributeValueId toAttributeValueId()
    {
        return this.value;
    }


    public String getAttributeId()
    {
        return this.value.getAttributeId();
    }

    public void setAttributeId(String attributeId)
    {
        this.value.setAttributeId(attributeId);
    }

    public String getValue()
    {
        return this.value.getValue();
    }

    public void setValue(String value)
    {
        this.value.setValue(value);
    }


    @Override
    public boolean equals(Object obj)
    {
        if (obj == this) {
            return true;
        }
        if (obj == null || obj.getClass() != this.getClass()) {
            return false;
        }

        AttributeValueIdFlattenedDto other = (AttributeValueIdFlattenedDto)obj;
        return value.equals(other.value);
    }

    @Override
    public int hashCode()
    {
        return value.hashCode();
    }

}

